enclave/CMakeLists.txt

```diff
@@ -32,61 +32,84 @@ elseif (OE_TRUSTZONE)
+      target_compile_options(${PARA_LIB_NAME} PUBLIC ${OE_TZ_TA_C_FLAGS})
+    endif ()
   endif ()
-endif ()
 
-enclave_enable_code_coverage(oeenclave)
+  enclave_enable_code_coverage(${PARA_LIB_NAME})
 
-# Add location of the oeedger8r-generated trusted headers for internal
-# functions implemented via EDL files.
-enclave_include_directories(oeenclave PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/core)
+  # Add location of the oeedger8r-generated trusted headers for internal
+  # functions implemented via EDL files.
+  enclave_include_directories(${PARA_LIB_NAME} PRIVATE
+                              ${CMAKE_CURRENT_BINARY_DIR}/core)
 
-enclave_link_libraries(oeenclave PUBLIC oelibc)
+  enclave_link_libraries(${PARA_LIB_NAME} PUBLIC oelibc)
 
-if (OE_TRUSTZONE)
-  enclave_link_libraries(oeenclave PUBLIC oelibutee)
-endif ()
+  if (OE_TRUSTZONE)
+    enclave_link_libraries(${PARA_LIB_NAME} PUBLIC oelibutee)
+  endif ()
 
-set_enclave_property(TARGET oeenclave PROPERTY ARCHIVE_OUTPUT_DIRECTORY
-                     ${OE_LIBDIR}/openenclave/enclave)
-install_enclaves(
-  TARGETS
-  oeenclave
-  EXPORT
-  openenclave-targets
-  ARCHIVE
-  DESTINATION
-  ${CMAKE_INSTALL_LIBDIR}/openenclave/enclave)
-
-if (WITH_EEID)
-  enclave_compile_definitions(oeenclave PRIVATE OE_WITH_EXPERIMENTAL_EEID)
-  # oeenclave currently does not link against the oecrypto* library by default.
-  # Consequently, it does not inherit the include paths of the crypto (e.g., mbedtls)
-  # headers, which are needed by eeid.c. Therefore, we add the paths here.
-  enclave_include_directories(
-    oeenclave PRIVATE
-    $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/3rdparty/mbedtls/mbedtls/include>
-    $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/3rdparty/mbedtls>)
-endif ()
+  if (NOT PARA_SKIP_INSTALL)
+    set_enclave_property(
+      TARGET ${PARA_LIB_NAME} PROPERTY ARCHIVE_OUTPUT_DIRECTORY
+      ${OE_LIBDIR}/openenclave/enclave)
+    install_enclaves(
+      TARGETS
+      ${PARA_LIB_NAME}
+      EXPORT
+      openenclave-targets
+      ARCHIVE
+      DESTINATION
+      ${CMAKE_INSTALL_LIBDIR}/openenclave/enclave)
+  endif ()
+
+  if (WITH_EEID)
+    enclave_compile_definitions(${PARA_LIB_NAME} PRIVATE
+                                OE_WITH_EXPERIMENTAL_EEID)
+    # oeenclave currently does not link against the oecrypto* library by default.
+    # Consequently, it does not inherit the include paths of the crypto (e.g., mbedtls)
+    # headers, which are needed by eeid.c. Therefore, we add the paths here.
+    enclave_include_directories(
+      ${PARA_LIB_NAME} PRIVATE
+      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/3rdparty/mbedtls/mbedtls/include>
+      $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}/3rdparty/mbedtls>)
+  endif ()
+
+  enclave_enable_fuzzing(${PARA_LIB_NAME})
+endmacro ()
 
-enclave_enable_fuzzing(oeenclave)
+add_library_oeenclave_wrapper(LIB_NAME oeenclave)
+# This lib should only be used to build oeutil
+add_library_oeenclave_wrapper(LIB_NAME oeenclave_loophole SKIP_INSTALL)
```

Comment:
I just realize that we could add more changes to this library version for other testing purposes. Let's rename the library to `oeenclave_prerelease_test`.